DB backups broken: KINDRED_DB_URL missing in secrets.env + pg16 vs pg17 version mismatch

## Root Cause Summary

All DB backups have been broken since ~early May 2026. There are **two independent bugs**:

---

### Bug 1: KINDRED_DB_URL (and LACHESIS_DB_URL) missing from secrets.env — BLOCKS ALL BACKUPS

**Impact:** Every cron run since the Kindred/Lachesis backup sections were added exits immediately at line 17 — no DBs are backed up at all.

**Details:**
- `~/.config/archon-cron/secrets.env` (last modified Apr 25 22:58) only contains: `ANNIE_DB_URL`, `RELI_DB_URL`, `FILMDUEL_DB_URL`
- The script added `KINDRED_DB_URL` and `LACHESIS_DB_URL` requirements (commits 456eef9, 9eb0835, 9e5a7b8, 43, 44) but the secrets file was never updated
- Due to `set -euo pipefail`, the `: "${KINDRED_DB_URL:?...}"` check at line 17 exits with error 1 before any backup runs
- The entire `/tmp/db-backup.log` tail is wall-to-wall `KINDRED_DB_URL not set` errors

**Fix needed:** Add `KINDRED_DB_URL` and `LACHESIS_DB_URL` to `~/.config/archon-cron/secrets.env` with the correct Supabase connection strings.

---

### Bug 2: pg_dump version mismatch — would block FilmDuel, Kindred, Lachesis, Annie, Reli even if Bug 1 is fixed

**Impact:** All pg_dump calls fail because the local client is pg16 but Supabase servers run pg17.

**Details:**
- Local: `pg_dump (PostgreSQL) 16.14 (Ubuntu 16.14-0ubuntu0.24.04.1)` at `/usr/bin/pg_dump`
- Remote: Supabase PostgreSQL 17.6 (confirmed on all tested DBs: Reli, FilmDuel)
- Error: `pg_dump: error: aborting because of server version mismatch — server version: 17.6; pg_dump version: 16.14`
- `postgresql-client-17` is NOT available via default Ubuntu 24.04 apt repos (requires PGDG PPA)
- The 131B FilmDuel "backup" files are gzipped error messages, not DB dumps — confirmed by `zcat`
- This bug started around Apr 20 (earliest log entry is `ERROR: FilmDuel pg_dump failed`) which aligns with Supabase upgrading to PG17

**Fix needed:** Install `postgresql-client-17` via the PGDG apt repository:
```bash
sudo apt-get install -y curl ca-certificates
curl -fsSL https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo gpg --dearmor -o /etc/apt/trusted.gpg.d/postgresql.gpg
echo "deb https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" | sudo tee /etc/apt/sources.list.d/pgdg.list
sudo apt-get update && sudo apt-get install -y postgresql-client-17
```
Then update the script to use the full path `/usr/lib/postgresql/17/bin/pg_dump` or ensure PATH includes it.

---

### Historical context on Annie/Reli backups

- Before Apr 20: Annie and Reli used `node scripts/export-to-sqlite.mjs` (export to SQLite) — this was failing because the script file didn't exist (`Cannot find module '…/scripts/export-to-sqlite.mjs'`)
- Commits c6ae65a + a7d1576 (late Apr) migrated them to pg_dump approach — but they immediately hit Bug 2 (version mismatch) and Bug 1 (missing KINDRED secret)
- The annie backup directory contains only an old pre-deploy dump from Apr 3 (0B file + a directory from an aborted run)
- The reli backup directory is completely empty

---

### Current state of backup files

| DB | Last real backup | Status |
|---|---|---|
| Annie | Apr 3 (pre-deploy only) | No backups since; directory contains 0B file |
| Reli | Never (pg_dump era) | Empty directory |
| FilmDuel | Never (all 131B files are gzipped error msgs) | Script runs but creates useless files |
| Kindred | Never | Directory not even created (script exits before mkdir) |
| Lachesis | Never | Directory not even created |

---

## Action Items

1. **Immediate:** Add `KINDRED_DB_URL` and `LACHESIS_DB_URL` to `~/.config/archon-cron/secrets.env`
2. **Immediate:** Install `postgresql-client-17` via PGDG PPA on the backup host
3. **Verify:** Run `backup-dbs.sh` manually after both fixes and confirm non-trivial backup sizes
4. **Consider:** The script should use `|| true` or remove `set -e` around the per-DB `:?` checks so one missing secret doesn't block all other DBs from backing up

--- a/ops/cron/README.md
+++ b/ops/cron/README.md
@@ -19,10 +19,21 @@ $ARCHON_CRON_SECRETS   (default: ~/.config/archon-cron/secrets.env, chmod 600)
 
 Required keys (see individual scripts for which ones each uses):
 
-- `ANNIE_DB_URL`, `RELI_DB_URL`, `FILMDUEL_DB_URL` — Supabase connection strings used by `backup-dbs.sh`.
+- `ANNIE_DB_URL`, `RELI_DB_URL`, `FILMDUEL_DB_URL`, `KINDRED_DB_URL`, `LACHESIS_DB_URL` — Supabase connection strings used by `backup-dbs.sh`. Missing entries cause that DB to be skipped (not a hard failure).
 - `NTFY_TOPIC` — private ntfy.sh topic for notifications. No fallback default; scripts fail loud if missing.
 
 Set perms: `chmod 600 ~/.config/archon-cron/secrets.env`.
+
+### pg_dump version (required for Supabase pg17)
+
+`backup-dbs.sh` auto-selects `/usr/lib/postgresql/17/bin/pg_dump` if present. Install it once:
+
+```bash
+sudo apt-get install -y curl ca-certificates
+curl -fsSL https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo gpg --dearmor -o /etc/apt/trusted.gpg.d/postgresql.gpg
+echo "deb https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" | sudo tee /etc/apt/sources.list.d/pgdg.list
+sudo apt-get update && sudo apt-get install -y postgresql-client-17
+```
 
 ## Install the crontab
 

--- a/ops/cron/backup-dbs.sh
+++ b/ops/cron/backup-dbs.sh
@@ -11,11 +11,14 @@ set -euo pipefail
 SECRETS_FILE="${ARCHON_CRON_SECRETS:-$HOME/.config/archon-cron/secrets.env}"
 # shellcheck source=/dev/null
 [ -r "$SECRETS_FILE" ] && . "$SECRETS_FILE"
-: "${ANNIE_DB_URL:?ANNIE_DB_URL not set — populate $SECRETS_FILE}"
-: "${RELI_DB_URL:?RELI_DB_URL not set — populate $SECRETS_FILE}"
-: "${FILMDUEL_DB_URL:?FILMDUEL_DB_URL not set — populate $SECRETS_FILE}"
-: "${KINDRED_DB_URL:?KINDRED_DB_URL not set — populate $SECRETS_FILE}"
-: "${LACHESIS_DB_URL:?LACHESIS_DB_URL not set — populate $SECRETS_FILE}"
+# Prefer postgresql-client-17 (matches Supabase pg17); fall back to whatever is in PATH.
+if [ -x /usr/lib/postgresql/17/bin/pg_dump ]; then
+    PG_DUMP=/usr/lib/postgresql/17/bin/pg_dump
+    PSQL=/usr/lib/postgresql/17/bin/psql
+else
+    PG_DUMP=$(command -v pg_dump)
+    PSQL=$(command -v psql)
+fi
 
 BACKUP_ROOT="/mnt/steam-slow/backups"
 TIMESTAMP=$(date +%Y%m%d-%H%M%S)
@@ -39,9 +42,11 @@ ANNIE_OUT="$ANNIE_BACKUP_DIR/annie-${TIMESTAMP}.sql.gz"
 # Note: --schema=annie and annie."Project" assume:
 #   1. 017-annie-schema-isolation.sql has been run against the consolidated DB
 #   2. ANNIE_DB_URL (in secrets.env) has been updated to the consolidated DB URL
-if pg_dump "$ANNIE_DB_URL" --no-owner --no-acl --schema=annie 2>&1 | gzip > "$ANNIE_OUT"; then
+if [ -z "${ANNIE_DB_URL:-}" ]; then
+    log "SKIP: Annie — ANNIE_DB_URL not set (populate $SECRETS_FILE)"
+elif $PG_DUMP "$ANNIE_DB_URL" --no-owner --no-acl --schema=annie 2>&1 | gzip > "$ANNIE_OUT"; then
     SIZE=$(du -h "$ANNIE_OUT" | cut -f1)
-    PROJECTS=$(psql "$ANNIE_DB_URL" -t -c "SELECT count(*) FROM annie.\"Project\"" 2>/dev/null | tr -d ' ' || echo "?")
+    PROJECTS=$($PSQL "$ANNIE_DB_URL" -t -c "SELECT count(*) FROM annie.\"Project\"" 2>/dev/null | tr -d ' ' || echo "?")
     log "Annie backed up: $ANNIE_OUT ($SIZE, $PROJECTS projects)"
     if [ "$PROJECTS" = "0" ]; then
         log "WARNING: Annie backup has 0 projects — possible data loss!"
@@ -55,9 +60,11 @@ RELI_BACKUP_DIR="$BACKUP_ROOT/reli"
 mkdir -p "$RELI_BACKUP_DIR"
 
 RELI_OUT="$RELI_BACKUP_DIR/reli-${TIMESTAMP}.sql.gz"
-if pg_dump "$RELI_DB_URL" --no-owner --no-acl --schema=reli | gzip > "$RELI_OUT"; then
+if [ -z "${RELI_DB_URL:-}" ]; then
+    log "SKIP: Reli — RELI_DB_URL not set (populate $SECRETS_FILE)"
+elif $PG_DUMP "$RELI_DB_URL" --no-owner --no-acl --schema=reli | gzip > "$RELI_OUT"; then
     SIZE=$(du -h "$RELI_OUT" | cut -f1)
-    THINGS=$(psql "$RELI_DB_URL" -t -c "SELECT count(*) FROM reli.things" 2>/dev/null | tr -d ' ' || echo "?")
+    THINGS=$($PSQL "$RELI_DB_URL" -t -c "SELECT count(*) FROM reli.things" 2>/dev/null | tr -d ' ' || echo "?")
     log "Reli backed up: $RELI_OUT ($SIZE, $THINGS things)"
     if [ "$THINGS" = "0" ] || [ "$THINGS" = "?" ]; then
         log "WARNING: Reli backup has 0 or unknown things — possible data loss or schema not yet migrated!"
@@ -74,9 +81,11 @@ FILMDUEL_OUT="$FILMDUEL_BACKUP_DIR/filmduel-${TIMESTAMP}.sql.gz"
 # Note: --schema=filmduel and filmduel.users assume:
 #   1. 014-filmduel-schema-isolation.sql has been run against the consolidated DB
 #   2. FILMDUEL_DB_URL (in Railway) has been updated to the consolidated DB URL
-if pg_dump "$FILMDUEL_DB_URL" --no-owner --no-acl --schema=filmduel | gzip > "$FILMDUEL_OUT"; then
+if [ -z "${FILMDUEL_DB_URL:-}" ]; then
+    log "SKIP: FilmDuel — FILMDUEL_DB_URL not set (populate $SECRETS_FILE)"
+elif $PG_DUMP "$FILMDUEL_DB_URL" --no-owner --no-acl --schema=filmduel | gzip > "$FILMDUEL_OUT"; then
     SIZE=$(du -h "$FILMDUEL_OUT" | cut -f1)
-    USERS=$(psql "$FILMDUEL_DB_URL" -t -c "SELECT count(*) FROM filmduel.users" 2>/dev/null | tr -d ' ' || echo "?")
+    USERS=$($PSQL "$FILMDUEL_DB_URL" -t -c "SELECT count(*) FROM filmduel.users" 2>/dev/null | tr -d ' ' || echo "?")
     log "FilmDuel backed up: $FILMDUEL_OUT ($SIZE, $USERS users)"
     if [ "$USERS" = "0" ]; then
         log "WARNING: FilmDuel backup has 0 users — possible data loss!"
@@ -90,9 +99,11 @@ KINDRED_BACKUP_DIR="$BACKUP_ROOT/kindred"
 mkdir -p "$KINDRED_BACKUP_DIR"
 
 KINDRED_OUT="$KINDRED_BACKUP_DIR/kindred-${TIMESTAMP}.sql.gz"
-if pg_dump "$KINDRED_DB_URL" --no-owner --no-acl --schema=kindred | gzip > "$KINDRED_OUT"; then
+if [ -z "${KINDRED_DB_URL:-}" ]; then
+    log "SKIP: Kindred — KINDRED_DB_URL not set (populate $SECRETS_FILE)"
+elif $PG_DUMP "$KINDRED_DB_URL" --no-owner --no-acl --schema=kindred | gzip > "$KINDRED_OUT"; then
     SIZE=$(du -h "$KINDRED_OUT" | cut -f1)
-    ENTRIES=$(psql "$KINDRED_DB_URL" -t -c "SELECT count(*) FROM kindred.entries" 2>/dev/null | tr -d ' ' || echo "?")
+    ENTRIES=$($PSQL "$KINDRED_DB_URL" -t -c "SELECT count(*) FROM kindred.entries" 2>/dev/null | tr -d ' ' || echo "?")
     log "Kindred backed up: $KINDRED_OUT ($SIZE, $ENTRIES entries)"
     if [ "$ENTRIES" = "0" ] || [ "$ENTRIES" = "?" ]; then
         log "WARNING: Kindred backup has 0 or unknown entries — possible data loss or schema not yet migrated!"
@@ -106,11 +117,13 @@ LACHESIS_BACKUP_DIR="$BACKUP_ROOT/lachesis"
 mkdir -p "$LACHESIS_BACKUP_DIR"
 
 LACHESIS_OUT="$LACHESIS_BACKUP_DIR/lachesis-${TIMESTAMP}.sql.gz"
-if pg_dump "$LACHESIS_DB_URL" --no-owner --no-acl --schema=lachesis | gzip > "$LACHESIS_OUT"; then
+if [ -z "${LACHESIS_DB_URL:-}" ]; then
+    log "SKIP: Lachesis — LACHESIS_DB_URL not set (populate $SECRETS_FILE)"
+elif $PG_DUMP "$LACHESIS_DB_URL" --no-owner --no-acl --schema=lachesis | gzip > "$LACHESIS_OUT"; then
     SIZE=$(du -h "$LACHESIS_OUT" | cut -f1)
     log "Lachesis backed up: $LACHESIS_OUT ($SIZE)"
     # TODO: once lachesis schema is populated, add row count check like FilmDuel/Kindred:
-    # ROWS=$(psql "$LACHESIS_DB_URL" -t -c "SELECT count(*) FROM lachesis.{primary_table}" 2>/dev/null | tr -d ' ' || echo "?")
+    # ROWS=$($PSQL "$LACHESIS_DB_URL" -t -c "SELECT count(*) FROM lachesis.{primary_table}" 2>/dev/null | tr -d ' ' || echo "?")
     # log "Lachesis backup: $ROWS rows in {primary_table}"
     # if [ "$ROWS" = "0" ]; then log "WARNING: Lachesis backup has 0 rows — possible data loss!"; fi
 else

